src/app/ngx-mask/mask.service.ts

```diff
@@ -92,15 +92,27 @@ export class MaskService {
     }
 
     if (element !== this.document.activeElement) {
-      this.clearIfNotMatchFn(element);
+      this.clearIfNotMatchFn(element, maskedInput);
     }
   }
 
-  public clearIfNotMatchFn(element: HTMLInputElement): void {
-    if (this.clearIfNotMatch === true && this.maskExpression.length
-      !== element.value.length) {
-      element.value = '';
-    }
+  public clearIfNotMatchFn(element: HTMLInputElement, maskedInput: string): void {
+    if (!this.clearIfNotMatch) { return; }
+    if (this.maskExpression.length === element.value.length) { return; }
+
+    let regex: string = '';
+    this.maskExpression.split('').forEach((v: string, i: number) => {
```

Comment:
could we use Array.prototype.forEach.call instead split('') ?

Reply:
I'll try that out when I get home from work.  I think since intellisense didn't pull it up for me, so I went with the easiest route.

Otherwise, if it doesn't work, I'll just loop through it using a for loop instead of splitting it then doing a forEach.

Reply:
Looks like a string doesn't have a forEach unless I convert it to an array.  For now, I just used a for loop and iterated through it.  I've also updated the formatting and used template string for the mentioned code.

Thanks